packages/studio-be/src/core/bots/bot-service.ts

```diff
@@ -63,7 +63,13 @@ const BotCreationSchema = Joi.object().keys({
       id: Joi.string()
     }
   },
-  locked: Joi.bool()
+  locked: Joi.bool(),
+  isCloudBot: Joi.bool().optional(),
+  cloud: Joi.object({
+    oauthUrl: Joi.string().uri(),
+    clientId: Joi.string().guid(),
+    clientSecret: Joi.string().guid()
```

Comment:
Don't think the client secret is a GUID (that would be weird). Maybe you could simply check that it's a certain length. Something like 10 chars. I don't know if that adds any value but knowing the exact size of the key here we could let the user know pretty quickly when its key is definitely invalid. What do you think?

Reply:
The client secret and client id are both UUIDv4, and `guid()` is the [joi alias](https://joi.dev/api/?v=17.5.0#stringguid---aliases-uuid) for validating them. I got all this info from @michaelmass in case you want to confirm 👍🏻